File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    person1 = [1,2,3,4,5]
    person2 = [2,1,2,3,2,4,2,5]
    person3 = [3,3,1,1,2,2,4,4,5,5]
    
    ans1 = []
    ans2 = []
    ans3 = []
    
    sum1, sum2, sum3 = 0, 0, 0
    
    for i in range(len(answers)):
        ans1.append(person1[i%len(person1)])
        ans2.append(person2[i%len(person2)])
        ans3.append(person3[i%len(person3)])
    
    for idx, n in enumerate(answers):
        if n == ans1[idx]:
            sum1 += 1
        if n == ans2[idx]:
            sum2 += 1
        if n == ans3[idx]:
            sum3 += 1
    maxval = max(sum1, sum2, sum3)
    if sum1 == maxval: answer.append(1)
    if sum2 == maxval: answer.append(2)
    if sum3 == maxval: answer.append(3)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
       
        int[] person1 = {1, 2, 3, 4, 5};
        int[] person2 = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] person3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        
        int[] scores = new int[3];
        for (int i = 0; i < answers.length; i++){
            if (answers[i] == person1[i%person1.length]) {scores[0]++;}
            if (answers[i] == person2[i%person2.length]) {scores[1]++;}
            if (answers[i] == person3[i%person3.length]) {scores[2]++;}
        }
        
        int max_score = Arrays.stream(scores).max().getAsInt();
        
        ArrayList<Integer> answer = new ArrayList<>();
        for(int j=0; j<scores.length; j++){
            if (scores[j] == max_score){
                answer.add(j+1); 
            }
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}